src/pybind/mgr/dashboard/CMakeLists.txt

```diff
@@ -1,126 +1,141 @@
-include(CMakeParseArguments)
-function(add_npm_command)
-  set(options NODEENV)
-  set(single_kw OUTPUT COMMENT WORKING_DIRECTORY)
-  set(multi_kw COMMAND DEPENDS)
-  cmake_parse_arguments(NC "${options}" "${single_kw}" "${multi_kw}" ${ARGN})
-  string(REPLACE ";" " " command "${NC_COMMAND}")
-  if(NC_NODEENV)
-    string(REGEX REPLACE "^(.*(npm|npx) .*)$" ". ${mgr-dashboard-nodeenv-dir}/bin/activate && \\1 && deactivate" command ${command})
-  endif()
-  string(REPLACE " " ";" command "${command}")
-  add_custom_command(
-    OUTPUT "${NC_OUTPUT}"
-    COMMAND ${command}
-    DEPENDS ${NC_DEPENDS}
-    WORKING_DIRECTORY "${NC_WORKING_DIRECTORY}"
-    COMMENT ${NC_COMMENT})
-endfunction(add_npm_command)
+if(WITH_MGR_DASHBOARD_FRONTEND)
```

Comment:
i think we moved this check into the upper cmake, see d3ac8b132d466bc758e28450236811b630f641dd